### ISSUE TITLE
no-invalid-this rule false positive

### ISSUE THREAD

**Issue by thedark1337** [eslint/eslint#3287]
I was running ESLint through my project and encountered an issue with the no-invalid-this rule. 

ESLint version 1.0.0 and the output: 
![screenshot_2](https://cloud.githubusercontent.com/assets/3173170/9092811/04dd13ac-3b6e-11e5-854f-745c2fad5176.png)

This is the code I was linting: https://ghostbin.com/paste/jfgjn

Log.apply(this, args); is where the error shows up.

I am using my own ESLint [config](https://github.com/thedark1337/eslint-config-thedark1337) , it extends off of [XO](https://github.com/sindresorhus/eslint-config-xo).

The rule causing the issue is `'no-invalid-this': 2` in XO config that I extended off of.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
Thank you for the report.

I can reproduce that. I will work on this today (JST).

**Comment by mysticatea** [comment]
For an first aid: puts the comment after `return`.: 

```js
"use strict";

function foo() {
    return /* @this Log */ function() { this; };
}
```
